web/src/timerender.ts

```diff
@@ -355,6 +388,26 @@ export function get_markdown_time_tooltip(reference: HTMLElement): DocumentFragm
     return "";
 }
 
+function update_recent_topics_timestamps(): void {
+    if (recent_view_util.is_visible()) {
+        // Don't update timestamps when the recent_topics view is hidden.
+        return;
+    }
+    for (const entry of recent_topics_update_list) {
+        const conversation_key = entry.conversation_key!;
+        const $element = $(
+            "#" + CSS.escape(`recent_conversation:${conversation_key}`) + " .last_msg_time",
+        );
+        if (!$element.length) {
+            blueslip.warn("Recent topic timestamp element not found.");
+            continue;
+        }
+
+        const rendered_time = relative_time_string_from_date({date: entry.time});
+        render_recent_topic_timestamp($element, rendered_time);
+    }
+}
```

Comment:
So this function is being called every minute and seems somewhat expensive -- at least it doesn't seem like we're trying to avoid doing a bunch of DOM operations in cases where the answers can't possibly have changed. We have a very nice check in the message feed `if (!isEqual(today, last_update))`; I wonder if something like that could be used here?

Reply:
We actually check if the new and old timestamps are the same in `render_recent_topic_timestamp`.
```javascript
if (!current_timestamp || current_timestamp === new_timestamp) {
    return;
}
```

Reply:
Yeah, but we're doing that check after already having looked up every row in the DOM to read the timestamps and calculate new ones, which feels expensive.

I don't know, maybe it's OK because these tables are not huge, but also the **vast majority** of the rows will be expected to never need an update except on the hour.